src/AugmentedMatrix.jl

```diff
@@ -0,0 +1,78 @@
+using LinearAlgebra, SparseArrays
+
+#Used in one line in function below
+function invert_or_zero(x::Real)
+    if x == 0
+        return 0
+    else
+        return 1 / x
+    end
+end
+
+"""
+    augmentedRateMatrix(rates_tensor::Array{T,3}, time_steps::Vector{T})
+
+Computes the augmented rate matrix from several rate matrices and a succession
+of time steps. The slices of `rates_tensor` are rate matrices that describe the
+rates of transition between the states of a Discretized Markov process at times
+for `t` in `time_steps`.
+
+# Notation:
+- `M`: is the number of disjoint intervals of time.
+- `N`: is the number of states of the Discretized Markov process.
+"""
+function augmentedRateMatrix(rates_tensor::Array{T,3}, time_steps::Vector{T}) where {T<:Real}
+    N, _, M = size(rates_tensor)
+
+    M == length(time_steps) || throw(ArgumentError("More rate matrices than " *
+                                                   "time steps were provided."))
+    M >= 1 || throw(ArgumentError("At least 2 time steps are needed for the analysis"))
+
+
+    # Pre-allocating the final matrix J
+    J = spzeros(T, N * M, N * M)
+
+    #q in the paper is this, but with a minus sign. 
+    #Though we never use the negative value so I didn't bother to invert this.
+    # FIXME: Given that this is a rate matrix, does this just reduce to the diagonal of the matrix?
```

Comment:
I believe this will be fixed in the follow up PR , but just a reminder to fix it and then remove the FIXME comment